Description: Fix credential leaking of keycloak_user
Origin: upstream, https://github.com/ansible-collections/community.general/pull/11005
Bug: https://github.com/ansible-collections/community.general/issues/11000
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/ansible_collections/community/general/changelogs/fragments/11005-keycloak_user.yml b/ansible_collections/community/general/changelogs/fragments/11005-keycloak_user.yml
new file mode 100644
index 0000000..d715ca8
--- /dev/null
+++ b/ansible_collections/community/general/changelogs/fragments/11005-keycloak_user.yml
@@ -0,0 +1,4 @@
+security_fixes:
+  - "keycloak_user - the parameter ``credentials[].value`` is now marked as ``no_log=true``. Before it was logged by Ansible, unless the task was marked as ``no_log: true``.
+     Since this parameter can be used for passwords, this resulted in credential leaking
+     (https://github.com/ansible-collections/community.general/issues/11000, https://github.com/ansible-collections/community.general/pull/11005)."
diff --git a/ansible_collections/community/general/plugins/modules/keycloak_user.py b/ansible_collections/community/general/plugins/modules/keycloak_user.py
index 2b3c838..ab038f9 100644
--- a/ansible_collections/community/general/plugins/modules/keycloak_user.py
+++ b/ansible_collections/community/general/plugins/modules/keycloak_user.py
@@ -357,7 +357,7 @@ def main():
     argument_spec['auth_username']['aliases'] = []
     credential_spec = dict(
         type=dict(type='str', required=True),
-        value=dict(type='str', required=True),
+        value=dict(type='str', required=True, no_log=True),
         temporary=dict(type='bool', default=False)
     )
     client_consents_spec = dict(
